File: Utility/ROMs/CheatData.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;

namespace Utility
{
    public sealed class CheatData : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged = delegate { };

        private string code;
        private string description;
        private bool enabled;

        public String CheatCode
        {
            get
            {
                return this.code;
            }
            set
            {
                if(value != code)
                {
                    this.code = value;
                    this.NotifyPropertyChanged();
                }
            }
        }

        public String Description
        {
            get
            {
                return this.description;
            }
            set
            {
                if (value != description)
                {
                    this.description = value;
                    this.NotifyPropertyChanged();
                }
            }
        }

        public bool Enabled
        {
            get
            {
                return this.enabled;
            }
            set
            {
                if (value != enabled)
                {
                    this.enabled = value;
                    this.NotifyPropertyChanged();
                }
            }
        }

        private void NotifyPropertyChanged([CallerMemberName] string propertyName = null)
        {
            this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}

File: EmulatorComponent/CheatData.cpp
#include "pch.h"
#include "CheatData.h"

using namespace Windows::UI::Xaml::Data;

namespace EmulatorComponent
{
	Platform::String ^CheatData::CheatCode::get()
	{
		return this->code;
	}

	void CheatData::CheatCode::set(Platform::String ^value)
	{
		if (this->code != value)
		{
			this->code = value;
		}
	}

	Platform::String ^CheatData::Description::get()
	{
		return this->description;
	}

	void CheatData::Description::set(Platform::String ^value)
	{
		if (this->description != value)
		{
			this->description = value;
		}
	}

	bool CheatData::Enabled::get()
	{
		return this->enabled;
	}

	void CheatData::Enabled::set(bool value)
	{
		if (this->enabled != value)
		{
			this->enabled = value;
		}
	}
}